Source/KSCrash/Recording/Tools/KSCxaThrowSwapper.c

```diff
@@ -0,0 +1,373 @@
+        {
+            uintptr_t function = findAddress(info.dli_fbase);
+            if (function != (uintptr_t) NULL)
+            {
+                cxa_throw_type original = (cxa_throw_type) function;
+                original(thrown_exception, tinfo, dest);
+            }
+        }
+    }
+}
+
+static vm_prot_t get_protection(void *sectionStart)
+{
+    mach_port_t task = mach_task_self();
+    vm_size_t size = 0;
+    vm_address_t address = (vm_address_t) sectionStart;
+    memory_object_name_t object;
+#if __LP64__
+    mach_msg_type_number_t count = VM_REGION_BASIC_INFO_COUNT_64;
+    vm_region_basic_info_data_64_t info;
+    kern_return_t info_ret =
+        vm_region_64(task, &address, &size, VM_REGION_BASIC_INFO_64, (vm_region_info_64_t) &info, &count, &object);
+#else
+    mach_msg_type_number_t count = VM_REGION_BASIC_INFO_COUNT;
+    vm_region_basic_info_data_t info;
+    kern_return_t info_ret =
+        vm_region(task, &address, &size, VM_REGION_BASIC_INFO, (vm_region_info_t)&info, &count, &object);
+#endif
+    if (info_ret == KERN_SUCCESS)
+    {
+        return info.protection;
+    }
+    else
+    {
+        return VM_PROT_READ;
+    }
+}
+
+static bool get_commands(const struct mach_header *header,
+                         const struct symtab_command **symtab_cmd,
+                         const struct dysymtab_command **dysymtab_cmd)
+{
+    segment_command_t *cur_seg_cmd = NULL;
+    uintptr_t cur = (uintptr_t) header + sizeof(mach_header_t);
+    
+    const struct symtab_command *local_symtab_cmd = NULL;
+    const struct dysymtab_command *local_dysymtab_cmd = NULL;
+    
+    for (uint i = 0; i < header->ncmds; i++, cur += cur_seg_cmd->cmdsize)
+    {
+        cur_seg_cmd = (segment_command_t *) cur;
+        switch (cur_seg_cmd->cmd)
+        {
+            case LC_SYMTAB:
+                local_symtab_cmd = (const struct symtab_command *) cur_seg_cmd;
+                if (symtab_cmd != NULL)
+                {
+                    *symtab_cmd = local_symtab_cmd;
+                }
+                break;
+            case LC_DYSYMTAB:
+                local_dysymtab_cmd = (const struct dysymtab_command *) cur_seg_cmd;
+                if (dysymtab_cmd != NULL)
+                {
+                    *dysymtab_cmd = local_dysymtab_cmd;
+                }
+                break;
+            default:
+                break;
+        }
+        
+        if (local_symtab_cmd != NULL &&
+            local_dysymtab_cmd != NULL &&
+            local_dysymtab_cmd->nindirectsyms != 0)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+static bool get_sections(const segment_command_t *data_seg,
+                         const section_t **lazy_sym_sect,
+                         const section_t **non_lazy_sym_sect)
+{
+    if (strcmp(data_seg->segname, SEG_DATA) == false ||
+        strcmp(data_seg->segname, SEG_DATA_CONST) == false)
```

Comment:
This feels weird comparing `strcmp` to `false`, since it's an integer return that is 0 if the args are the same. Also it's compared to `0` in other parts of the code.